src/System.Private.CoreLib/src/System/ByReference.cs

```diff
@@ -8,8 +8,8 @@ namespace System
 {
     // ByReference<T> is meant to be used to represent "ref T" fields. It is working
     // around lack of first class support for byref fields in C# and IL. The JIT and 
-    // type loader have special handling for it that turns it into a thin wrapper around ref T.
-    [System.Runtime.CompilerServices.DependencyReductionRoot] // TODO: put this in System.Private.ILToolchain contract instead
+    // type loader has special handling for it that turns it into a thin wrapper around ref T.
+    [DependencyReductionRoot] // TODO: put this in System.Private.ILToolchain contract instead
```

Comment:
You can add `#if PROJECTN` around this like we have it in other places for this attribute; and delete the TODO.